src/index.js

```diff
@@ -65,6 +65,35 @@ function areSetsEqual(set1, set2) {
   return !Array.from(set1).some(item => !set2.has(item))
 }
 
+function memberExpressionRootIdentifier(path) {
+  // Traverse up to the parent before the topmost member expression, and then
+  // traverse back down to find the topmost identifier. It seems like there
+  // might be a better way to do this.
+  const parent = path.findParent(p => !p.isMemberExpression())
```

Comment:
will this also handle the case where you have `foo.bar.baz`?

Reply:
nvm, I see that in your tests now.

Reply:
Yes, I believe it will. Here's the test case for that: https://github.com/oliviertassinari/babel-plugin-transform-react-remove-prop-types/pull/158/files#diff-1aea5e60442f49f6365c94f57556c51aR17

Reply:
I updated this PR so that the somePropTypes objects aren't shared between the components, which should give us more confidence in each individual instance being properly cleaned up.